### PR TITLE
Python client Guide

### PR DESCRIPTION
**Pull request by abidlabs**
Adds a Guide for the Python client

@aliabd can we add a section for the client-related guides on the Guides page?

### CODE REVIEW

**Comment by aliabd** [comment]
A bit confused @abidlabs it looks like you already added the section. Creating a folder is all that's needed :) 
Screen Shot 2023-04-08 at 11 11 46 PM

File: guides/06_client-libraries/01_getting-started-with-the-python-client.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+from gradio_client import Client
+
+client = Client("https://bec81a83-5b5c-471e.gradio.live")
+```
+
+### Inspecting the API endpoints
+
+Once you have connected to a Gradio app, you can view the APIs that are available to you by calling the `Client.view_api()` method. For the Whisper Space, we see the following:
+
+```
```
Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:74]: Code blocks without a language break rendering on the website. 

Screen Shot 2023-04-08 at 11 16 04 PM

Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:74]:
```suggestion
```bash
```

File: guides/06_client-libraries/01_getting-started-with-the-python-client.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+Otherwise, the lightweight `gradio_client` package can be installed from pip (or pip3) and is tested to work with Python versions 3.9 or higher:
+
+```bash
+$ pip install gradio_client
+```
+
+## Usage
+
+### Connecting to a running Gradio App
```
Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:39]: right now only the headers with ## are showing on side nav

Screen Shot 2023-04-08 at 11 24 55 PM

Better to show all these headers

Screen Shot 2023-04-08 at 11 26 54 PM 

also some headers are just bold and not headers like 

> `**Connecting a general Gradio app**`

I suggest making all of these ## headers

Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:39]:
```suggestion
## Connecting to a running Gradio App
```

File: guides/06_client-libraries/01_getting-started-with-the-python-client.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+If you already have a recent version of `gradio`, then the `gradio_client` is included as a dependency. 
+
+Otherwise, the lightweight `gradio_client` package can be installed from pip (or pip3) and is tested to work with Python versions 3.9 or higher:
+
+```bash
+$ pip install gradio_client
+```
+
+## Usage
```
Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:37]:
```suggestion
```

File: guides/06_client-libraries/01_getting-started-with-the-python-client.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+### Connecting to a running Gradio App
+
+Start by connecting instantiating a `Client` object and connecting it to a Gradio app that is running on Hugging Face Spaces or generally anywhere on the web.
+
+**Connecting to a Hugging Face Space**
```
Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:43]:
```suggestion
## Connecting to a Hugging Face Space
```

File: guides/06_client-libraries/01_getting-started-with-the-python-client.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+You can also connect to private Spaces by passing in your HF token with the `hf_token` parameter. You can get your HF token here: https://huggingface.co/settings/tokens
+
+```python
+from gradio_client import Client
+
+client = Client("abidlabs/my-private-space", hf_token="...") 
+```
+
+
+**Connecting a general Gradio app**
```
Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:60]:
```suggestion
## Connecting a general Gradio app
```

File: guides/06_client-libraries/01_getting-started-with-the-python-client.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+If your app is running somewhere else, just provide the full URL instead, including the "http://" or "https://". Here's an example of making predictions to a Gradio app that is running on a share URL:
+
+```python
+from gradio_client import Client
+
+client = Client("https://bec81a83-5b5c-471e.gradio.live")
+```
+
+### Inspecting the API endpoints
```
Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:70]:
```suggestion
## Inspecting the API endpoints
```

File: guides/06_client-libraries/01_getting-started-with-the-python-client.md [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+    Returns:
+     - [Textbox] value_0: str (value)
+```
+
+This shows us that we have 1 API endpoint in this space, and shows us how to use the API endpoint to make a prediction: we should call the `.predict()` method (which we will explore below), providing a parameter `input_audio` of type `str`, which is a `filepath or URL`. 
+
+We should also provide the `api_name='/predict'` argument to the `predict()` method. Although this isn't necessary if a Gradio app has only 1 named endpoint, it does allow us to call different endpoints in a single app if they are available. If an app has unnamed API endpoints, these can also be displayed by running `.view_api(all_endpoints=True)`.
+
+
+### Making a prediction
```
Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:91]:
```suggestion
## Making a prediction
```

File: guides/06_client-libraries/01_getting-started-with-the-python-client.md [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+from gradio_client import Client
+
+client = Client("abidlabs/whisper")
+client.predict("https://audio-samples.github.io/samples/mp3/blizzard_unconditional/sample-0.mp3")
+
+>> "My thought I have nobody by a beauty and will as you poured. Mr. Rochester is serve in that so don't find simpus, and devoted abode, to at might in a r—"
+```
+
+
+**Running jobs asyncronously**
```
Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:128]:
```suggestion
## Running jobs asyncronously
```

File: guides/06_client-libraries/01_getting-started-with-the-python-client.md [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+job = client.submit("Hello", api_name="/predict")  # This is not blocking
+
+# Do something else
+
+job.result()  # This is blocking
+
+>> Bonjour
+```
+
+**Adding callbacks**
```
Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:147]:
```suggestion
## Adding callbacks
```

File: guides/06_client-libraries/01_getting-started-with-the-python-client.md [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+
+job = client.submit("Hello", api_name="/predict", result_callbacks=[print_result])
+
+# Do something else
+
+>> The translated result is: Bonjour
+
+```
+
+**Status**
```
Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:167]:
```suggestion
## Status
```

File: guides/06_client-libraries/01_getting-started-with-the-python-client.md [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+client = Client(src="gradio/calculator")
+job = client.submit(5, "add", 4, api_name="/predict")
+job.status()
+
+>> <Status.STARTING: 'STARTING'>
+```
+
+The `Job` object also has a `done()` instance method which returns a boolean indicating whether the job has completed.
+
+### Generator Endpoints
```
Comment (aliabd) [inline guides/06_client-libraries/01_getting-started-with-the-python-client.md:183]:
```suggestion
## Generator Endpoints
```

**Comment by abidlabs** [comment]
> A bit confused @abidlabs it looks like you already added the section. Creating a folder is all that's needed :) 

Nice!

**Comment by abidlabs** [comment]
Thanks for the review and suggestions @aliabd! I've accepted the suggestions, but let's hold off on merging this Guide until we release the next version of the `gradio` library with the client.

**Comment by abidlabs** [comment]
Confirmed that the formatting is fixed now: 

image

**Pull request merged**